### PR TITLE
UI

### PR DESCRIPTION
**Pull request by StarMindz**
I created the App class with all needed function to implement a UI for the Ruby app

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    select_person = gets.chomp.to_i
+    case select_person
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'invalid input. Input either 1 or 2 according to your choice'
+    end
+  end
```
Comment (cynthiainga) [inline app.rb:70]: - [ ]  Good job so far but you still have some issues that need to be fixed. When I try to create a person (student or teacher) the console app failed with the message `wrong number of arguments`. Please kindly fix that 🙃 🙃

Screen Shot 2023-01-07 at 18 41 05

- [ ] Please after solving the bug you will make sure that the app works correctly 🙃 😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @StarMindz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Correct Giflow ✔️ 
- [ ] Good commit history ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comment under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by StarMindz** [comment]
@cynthiainga Thank you for the review, I've implemented your requested change

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require_relative './app'
+
+def main
+  new_app = App.new
+  new_app.run
+end
+
```
Comment (BenMukebo) [inline main.rb:7]: - [OPTIONAL] The `main.rb` file is the entry point that is invoked at the end of the file. You should Present the user with a list of options to perform here and not in the app.js file.
- This method should do and look similar to the following:

```ruby
def main
  app = App.new()

  response = nil

  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    ...
    puts "7 - Exit"
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
       ...
    when "7"
      # *Have a way to quit the app.
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

main()
```

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+    print 'Person id: '
+    person_id = gets.chomp.to_i
+
+    @rentals.each do |rental|
+      if rental.person.id.to_i == person_id
+        puts "Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
+      end
+    end
+  end
```
Comment (BenMukebo) [inline app.rb:171]: - [ ] Your school library console app must behave in the same way as the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) it was provided for you. Great job so far 👏🏻From Steps one up to five works fine, however, the last one step six has some mistakes that need to be addressed. Here is the error message I get when I list all rentals for a given person id 👇🏻 Please correct this.

Screenshot from 2023-01-07 19-20-21

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @StarMindz, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @StarMindz ,
Good job so far! 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- Your project looks awesome:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:



 
 ## Optional suggestions
 
 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Pull request merged**